### ISSUE TITLE
Add toggle to download from cache

### ISSUE THREAD

**Issue by lnp5131** [qsniyg/maxurl#1239]
It seems like it is becoming more common that it would open a new tab with the image and encounter a 403/404 error (resulting in null undefined in console), which cannot be saved but open in a popup fine and is present in cache. Cache seems to only be used for opening up the popup again without reloading the image.

**Comment by qsniyg** [comment]
Good idea! I think it will only work if "Allow showing partially loaded" is disabled for the target media type though.

**Comment by lnp5131** [comment]
Agreed, I wouldn't expect it to work with partially loaded or at least it would have to queue/await. To add to the post, I think sites are closing off "direct" links to their assets/images like Pixiv, Sankaku Complex, or wallpaper/stock photo sites. I'm not sure if maxurl handles the logic differently in the popup or is regarded as an additional iframe within the site and ties in with the cross-site vs same-site or if the expire headers are intervening. So it seems like downloading from cache might be a good medium to handle that.

**Comment by qsniyg** [comment]
@lnp5131 Have you tried using the download key? (bound to S/Ctrl+S by default)

**Comment by lnp5131** [comment]
That is what I was referring to. It always seems to open a new tab (firefox extension)
